gallery_dl/extractor/discord.py

```diff
@@ -0,0 +1,370 @@
+        text_content = [message["content"]]
+
+        for embed in message["embeds"]:
+            if embed["type"] == "rich":
+                text_content.append(embed.get("author", {}).get("name", ""))
+                text_content.append(embed.get("title", ""))
+                text_content.append(embed.get("description", ""))
+
+                for field in embed.get("fields", []):
+                    text_content.append(field.get("name", ""))
+                    text_content.append(field.get("value", ""))
+
+                text_content.append(embed.get("footer", {}).get("text", ""))
+
+        if message.get("poll"):
+            text_content.append(message["poll"]["question"]["text"])
+            for answer in message["poll"]["answers"]:
+                text_content.append(answer["poll_media"]["text"])
+
+        return "\n".join(t for t in text_content if t)
+
+    def extract_message(self, message):
+        # https://discord.com/developers/docs/resources/message#message-object-message-types
+        if message["type"] in (0, 19, 21):
+            message_metadata = {
+                **self.server_metadata,
+                **self.server_channels_metadata[message["channel_id"]],
+                "author": message["author"]["username"],
+                "author_id": message["author"]["id"],
+                "author_files": [],
+                "message": self.extract_message_text(message),
+                "message_id": message["id"],
+                "date": text.parse_datetime(
+                    message["timestamp"], "%Y-%m-%dT%H:%M:%S.%f%z"
+                ),
+                "files": []
+            }
+
+            for icon_type, icon_path in (
+                ("avatar", "avatars"),
+                ("banner", "banners")
+            ):
+                if message["author"].get(icon_type):
+                    message_metadata["author_files"].append({
+                        "url": self.cdn_fmt.format(
+                            icon_path,
+                            message_metadata["author_id"],
+                            message["author"][icon_type]
+                        ),
+                        "filename": icon_type,
+                        "extension": "png",
+                    })
+
+            for attachment in message["attachments"]:
+                message_metadata["files"].append({
+                    "url": attachment["url"],
+                    "type": "attachment"
+                })
+
+            for embed in message["embeds"]:
+                if embed["type"] in self.enabled_embeds:
+                    for field in ("video", "image", "thumbnail"):
+                        url = embed.get(field, {}).get("proxy_url")
+                        if url is not None:
+                            message_metadata["files"].append({
+                                "url": url,
+                                "type": "embed"
+                            })
+                            break
```

Comment:
Try to avoid the `dict.get("key", {})` pattern if possible. I think the following would be better:

```suggestion
                        if field not in embed:
                            continue
                        url = embed["field"].get("proxy_url")
                        if url is not None:
                            message_metadata["files"].append({
                                "url": url,
                                "type": "embed",
                            })
                            break
```